### PR TITLE
Regenerate nested Ids on character make and duplicate functions

### PR DESCRIPTION
**Pull request by lynn0702**
## ✅ Changes

**fix**
-

## 🌄 Context

Duplicated and made from template characters are currently retaining Ids from their original sheets which is causing collisions. 

## 🔒Checklist

- I tested my work on the feature environment

### CODE REVIEW

File: lib/domains/character/CharacterFactory.tsx [hunk 1]
```diff
@@ -78,12 +106,40 @@ export const CharacterFactory = {
...
+            return {
+              ...row,
+              columns: row.columns.map((column) => {
+                return {
+                  ...column,
+                  sections: column.sections.map((section) => {
+                    return {
+                      ...section,
+                      id: Id.generate(),
+                      pages: section.blocks.map((block) => {
```
Comment (RPDeshaies) [inline lib/domains/character/CharacterFactory.tsx:127]: **must**: i think here its supposed to be `blocks`

File: lib/domains/character/CharacterFactory.tsx [hunk 2]
```diff
@@ -37,6 +37,34 @@ export const CharacterFactory = {
     return {
       ...newCharacter,
       id: Id.generate(),
+            pages: newCharacter.pages.map((page) => {
```
Comment (RPDeshaies) [inline lib/domains/character/CharacterFactory.tsx:40]: nit: I think there's a formatting issue here

File: lib/domains/character/CharacterFactory.tsx (hunk 1 above)
Comment (lynn0702) [reply lib/domains/character/CharacterFactory.tsx:127]: You're right. I missed that edit moving from local to here.

File: lib/domains/character/CharacterFactory.tsx (hunk 2 above)
Comment (lynn0702) [reply lib/domains/character/CharacterFactory.tsx:40]: I'm not seeing it, but I'm happy to correct.

**Pull request merged**